projects/robots/universal_robots/resources/ros_package/ur_e_webots/launch/ur5e.launch

```diff
@@ -10,7 +11,7 @@
   <arg name="max_payload"  default="10.0" doc="upper payload bound for the robot" />
 
   <!-- send robot urdf to param server -->
-  <include file="$(find ur_e_description)/launch/ur5e_upload.launch">
+  <include file="$(find ur_e_description)/launch/ur5e_upload.launch" if="$(eval '1' if find('ur_e_description') else '0')">
```

Comment:
Unfortunately this doesn't seem to work for me.
I still get this error when running `roslaunch ur_e_webots ur5e.launch`:
```
stefi@libeccio:~/catkin_ws$ roslaunch ur_e_webots ur5e.launch 
... logging to /home/stefi/.ros/log/b568ee56-c117-11ea-a90a-00d8611a4965/roslaunch-libeccio-32108.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

Resource not found: ur_e_description
ROS path [0]=/opt/ros/melodic/share/ros
ROS path [1]=/home/stefi/catkin_ws/src
ROS path [2]=/opt/ros/melodic/share
The traceback for the exception was written to the log file
```

Instead if I remove this include, it seems the simulation runs and I only get these errors in the console:
```
stefi@libeccio:~/catkin_ws$ roslaunch ur_e_webots ur5e.launch 
... logging to /home/stefi/.ros/log/2ba39508-c118-11ea-a90a-00d8611a4965/roslaunch-libeccio-32201.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://libeccio:33945/

SUMMARY
========

PARAMETERS
 * /prefix: 
 * /robot_state_publisher/publish_frequency: 50.0
 * /robot_state_publisher/tf_prefix: 
 * /rosdistro: melodic
 * /rosversion: 1.14.6
 * /use_sim_time: True

NODES
  /
    fake_joint_calibration (rostopic/rostopic)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    webots_universal_robots (ur_e_webots/universal_robots_ros.py)

auto-starting new master
process[master]: started with pid [32216]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 2ba39508-c118-11ea-a90a-00d8611a4965
process[rosout-1]: started with pid [32228]
started core service [/rosout]
process[robot_state_publisher-2]: started with pid [32234]
process[fake_joint_calibration-3]: started with pid [32236]
process[webots_universal_robots-4]: started with pid [32237]
[ERROR] [1594212015.011784211]: Could not find parameter robot_description on parameter server
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of Webots will drop support for Python 2.7.
[robot_state_publisher-2] process has died [pid 32234, exit code 1, cmd /opt/ros/melodic/lib/robot_state_publisher/robot_state_publisher __name:=robot_state_publisher __log:=/home/stefi/.ros/log/2ba39508-c118-11ea-a90a-00d8611a4965/robot_state_publisher-2.log].
log file: /home/stefi/.ros/log/2ba39508-c118-11ea-a90a-00d8611a4965/robot_state_publisher-2*.log

```

Reply:
Ok, then it seems it is not working the same way on Kinetic and Melodic, as on Kinetic it seems to be working fine even if the package is not installed, I will try on a VM with Melodic.

Reply:
I can indeed reproduce with ros melodic :smiley:

Reply:
I tested and looked at many ways to do it but apparently, this is not well supported in ROS1 launch files.
The only possible way to do it seems to create a custom shell script that tests if the package is installed and to interprets the returned value from the launch file.
But this seems to be recommended and looked a bit overkill to me (we don't want to implement such mechanism for ROS1 while it works out of the box for ROS2, and in any case using these launch files does not make sense if the `ur_e_description` is not installed as both RVIz and Movit will not work).
So I would go for documenting this constraint and remove the not working changes to the launch file. @stefaniapedrazzi do you agree or have another solution to propose?

Reply:
Yes, I agree.
Documenting it seems fine.